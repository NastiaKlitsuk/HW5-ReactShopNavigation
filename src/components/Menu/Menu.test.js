import React from 'react';
import ReactDOM from 'react-dom';
import { Menu } from './index';
import { mount, shallow } from 'enzyme';
import {
  BrowserRouter as Router,
  Route
} from 'react-router-dom';
import expect from 'expect';
import renderer from 'react-test-renderer';

const socialLinks = [
  {
    text: 'facebook',
    icon: '',
    link: 'www.facebook.com'
  }]

const navigationLinks = [
  {
    name: 'Home',
    path: '/'
  }
]

const props = {
  children: jest.children,
  menuConfig: { socialLinks: socialLinks },
  isLoggedIn: false,
  navigationLinks: navigationLinks
}

const mountedMenu = mount(<Router><Route><Menu {...props} /></Route></Router>);

describe('Menu renders', () => {
  it('mounts', () => {
    mountedMenu
  })

  it('naviagtion links are rendered', () => {
    mountedMenu;
    const navigationLinksWrapper = mountedMenu.find('div.links-container');
    const linkWrapper = navigationLinksWrapper.find('Link').find('.page-link');
    expect(linkWrapper.at(0).text()).toBe('Home');
  })

  it('login link displayed when user did not log in', () => {
    const shallowMenu = shallow(<Menu {...props} />);
    const loginLink = shallowMenu.find('NavLink').find('.login');
    const shoppingCartLink = shallowMenu.find('NavLink').find('.shoppingCart');
    expect(loginLink.length).toBe(1);
    expect(shoppingCartLink.length).toBe(0);
  })

  it('shopping cart link displayed when user logged in', () => {
    const loginProps = {
      children: jest.children,
      menuConfig: { socialLinks: socialLinks },
      isLoggedIn: true,
      navigationLinks: navigationLinks
    }
    const shallowMenu = shallow(<Menu {...loginProps} />);
    const loginLink = shallowMenu.find('NavLink').find('.login');
    const shoppingCartLink = shallowMenu.find('NavLink').find('.shoppingCart');
    expect(loginLink.length).toBe(0);
    expect(shoppingCartLink.length).toBe(1);
  })

  it('menu renders correctly', () => {
    const tree = renderer.create(<Router><Route><Menu {...props} /></Route></Router>).toJSON();
    expect(tree).toMatchSnapshot();
  })
})
