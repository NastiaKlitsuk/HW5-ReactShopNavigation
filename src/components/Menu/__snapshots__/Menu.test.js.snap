// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu renders menu renders correctly 1`] = `
<div
  className="App-menubar"
>
  <div>
    <label
      className="hamburger-icon fa fa-bars"
      onClick={[Function]}
    />
    <div
      className="menu-content"
    >
      <div
        className="links-container"
      >
        <a
          className="page-link"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
        <a
          className="page-link"
          href="/products?referral=Amazon"
          onClick={[Function]}
        >
          Referral Link
        </a>
      </div>
      <div
        className="socialLinks"
      >
        <div
          className="social-links"
        >
          <a
            className="fa fa-"
            href="www.facebook.com"
            target="_blank"
          />
        </div>
      </div>
    </div>
    <div
      className="overlay"
    />
  </div>
  <div>
    <a
      aria-current={false}
      className="login"
      href="/login"
      onClick={[Function]}
      style={undefined}
    >
      Login
    </a>
  </div>
</div>
`;
